lib/dao.js

```diff
@@ -1206,7 +1206,7 @@ DataAccessObject._coerce = function (where) {
           operator = 'regexp';
         } else if (operator === 'regexp' && val instanceof RegExp) {
           // Do not coerce regex literals/objects
-        } else if (!((operator === 'like' || operator === 'nlike') && val instanceof RegExp)) {
+        } else if (!((operator === 'like' || operator === 'nlike') && val instanceof Object)) {
```

Comment:
As I have said in description, current loopback's `where` filter doesn't support complicated query which contains something mongoDB operators in an object.